src/foam/nanos/notification/email/DAOResourceLoader.java

```diff
@@ -32,38 +32,50 @@
 {
   protected String groupId_;
   protected String locale_;
+  protected String spid_;
 
-  public static EmailTemplate findTemplate(X x, String name, String groupId, String locale) {
+  public static EmailTemplate findTemplate(X x, String name, String groupId, String locale, String spid) {
     DAO groupDAO = (DAO) x.get("groupDAO");
     DAO emailTemplateDAO = (DAO) x.get("localEmailTemplateDAO");
-
-    /*
-    TODO:
-
-    name  group locale spid
-      Y     Y     Y     Y
-      Y     Y     Y     *
-      Y     Y     *     Y
-      Y     Y     *     *
-      Y     *     *     Y
-      Y     *     *     *
-    */
+    String group_ = ((Group) groupDAO.find(groupId)).getParent();
+    group_ = SafetyUtil.isEmpty(group_) ? groupId : group_;
```

Comment:
if your looking for the group root parent - which i think makes sense - you'll need this logic in a while loop.
One level up doesn't garantee the root parent.

Reply:
updated